### ISSUE TITLE
Implement email conversion with inner attached emails

### ISSUE THREAD

**Issue by nedashley** [bbottema/simple-java-mail#217]
Hi,

I have a use case where users are sending emails to be converted but the emails have inner message/rfc822 parts:

```
--000000000000f842de058e56eeb7
Content-Type: message/rfc822
Content-Disposition: attachment
Content-Transfer-Encoding: base64
X-Attachment-Id: 20028ebc3ab51dff_0.1
```

This currently on 5.2.0 results in a NullPointerException - line 129 of MimeMessageParser. The workaround I have is adding this slightly before line 129:

```
} else if (isMimeType(currentPart, "message/rfc822")) {
    final DataSource ds = createDataSource(currentPart);
    String attachmentId = parseXAttachmentID(currentPart);
    parsedComponents.attachmentList.put(valueNullOrEmpty(attachmentId) ? "Attachment" : String.format("Attachment - %s", attachmentId), ds);
} else {
```

This effectively allows the email to be seen as another attachment which can then be passed to the EmailConverter again.

Thanks
Ned

**Comment by bbottema** [comment]
Hmm, can you possibly provide an example message you verified reproduces the null pointer? I would like to test with the upcoming 6.0.0 as well.

**Comment by bbottema** [comment]
Any update here, @nedashley?

**Comment by nedashley** [comment]
Hi,

Sorry for the delay with this, please try this Test email that I have.

[Test Email.txt](https://github.com/bbottema/simple-java-mail/files/3486392/Test.Email.txt)

Thanks
Ned

**Comment by bbottema** [comment]
Hmm, I'm parsing this just fine with version 5.3.0, no NPE here. Can you please try with 5.3.0?

**Comment by bbottema** [comment]
No further feedback, I'm assuming it's fixed in 5.3.0. Closing for now.
